### ISSUE TITLE
Unable to build project 0.214.2 on Debian 11

### ISSUE THREAD

**Issue by bracci-quest-it** [n8n-io/n8n#5398]
**Describe the bug**
I have done a fresh installation of n8n in a fresh machine with Debian 11, I installed nvm, node16.19, enabled pnpm and cloned n8n.
`pnpm install` went well without errors; but on `pnpm build` I received this error:
```
n8n-editor-ui:build: cache miss, executing a4ebe2f9cb977c76
n8n-editor-ui:build: 
n8n-editor-ui:build: > n8n-editor-ui@0.180.2 build /home/.../n8n/packages/editor-ui
n8n-editor-ui:build: > cross-env VUE_APP_PUBLIC_PATH="/{{BASE_PATH}}/" NODE_OPTIONS="--max-old-space-size=8192" vite build
n8n-editor-ui:build: 
n8n-editor-ui:build: vite v4.0.4 building for production...
n8n-editor-ui:build: transforming...
n8n-editor-ui:build: ✓ 3099 modules transformed.
n8n-editor-ui:build:  ELIFECYCLE  Command failed with exit code 1.
n8n-editor-ui:build: ERROR: command finished with error: command (/home/.../n8n/packages/editor-ui) pnpm run build exited (1)
command (/home/.../n8n/packages/editor-ui) pnpm run build exited (1)
```

**To Reproduce**
`git clone https://github.com/n8n-io/n8n.git`
`git checkout n8n@0.214.2`
`pnpm install`
`pnpm build`

**Expected behavior**
It should build all packages or at least print an explicit error.

**Environment:**
- **OS** Debian 11
- **n8n** I tried both version `0.212.2` and version `0.214.2`
- **Node.js** I tried both version `18.13.0` and version `16.19.0`
- **npm** version `8.19.3`
- **pnpm** version `7.18.1`
- **Database** system: default
- **Operation** mode: default

**Comment by Joffcom** [comment]
Hey @bracci-quest-it,

Is there more to the output?

**Comment by bracci-quest-it** [comment]
@Joffcom  no, i also tried to check the verbose log of pnpm. I can provide the full stack trace and the verbose log file if needed.

**Comment by Joffcom** [comment]
Hey @bracci-quest-it,

If you can share the full output / log that could be very helpful, I have built n8n a few times today and I am not seeing the same issue so I suspect there was just something odd along the way. I did just give it a quick go on a small Ubuntu 20.04 server following the instructions you provided and I was not able to reproduce the same issue so I don't think it is a general n8n issue which would point to an oddity in Debian 11 or something environmental maybe a memory limit?

**Comment by bracci-quest-it** [comment]
Hello @Joffcom, I don't know how, but this morning (~4h ago) the project build went fine.
Since then I tried to start the CLI; both from CLI using `pnpm dev` and from VS CODE using the launch configuration `"Launch n8n CLI dev with debug"`.

In both cases the process ends without much info with the message:
`[nodemon] app crashed - waiting for file changes before starting...`.

Using the VS CODE debugger I think that the code crashes when executing [line 64 of `LoadNodesAndCredentials.ts`](https://github.com/n8n-io/n8n/blob/n8n%400.212.0/packages/cli/src/LoadNodesAndCredentials.ts#L64).
I also tried to change the `GENERATED_STATIC_DIR` constant in the file [`constants.ts`](https://github.com/n8n-io/n8n/blob/n8n%400.212.0/packages/cli/src/constants.ts#L22) but nothing changed.
The `$(whoami)` is in sudoers and it owns the folder `/home/${whoami}/.cache/n8n/public`, so it shouldn't be a permission problem.

This is the full console output, with the ENV property `"N8N_LOG_LEVEL"` set as `"verbose"`.
```
11:53:42 AM - Starting compilation in watch mode...
[TypeScript] [0] 
[Node] 
[Node] > n8n@0.212.0 start:default
[Node] > cd bin && ./n8n
[Node] 
[TypeScript] [1] tsc-alias info: [Watching for file changes...]
[Node] Debugger attached.
[Node] 2023-02-08T10:53:49.870Z | info     | Initializing n8n process "{ file: 'start.js', function: 'run' }"
[Node] Waiting for the debugger to disconnect...
[Node] Waiting for the debugger to disconnect...
[Node] Waiting for the debugger to disconnect...
[Node] Waiting for the debugger to disconnect...
[Node] [nodemon] app crashed - waiting for file changes before starting...
[TypeScript] [0] 
[TypeScript] [0] 11:53:54 AM - Found 0 errors. Watching for file changes.
```

Do you have any idea?
I am willing to format and change OS, if that can help; in order to replicate your environment as much as possible.
What OS does the dev team use? What specific version of node, npm and pnpm?

**Comment by Joffcom** [comment]
Hey @bracci-quest-it,

It sounds like the issue is not related to the OS at all is possibly just something else going on, The good news is n8n appears to be working as expected.

I would maybe make sure you are using node 16 still and use the latest release or better the master branch as it could be that we have already fixed something.

**Comment by martinse** [comment]
Had the same issue as the first comment here with the error 

```
[...]
n8n-editor-ui:build:  ELIFECYCLE  Command failed with exit code 1.
n8n-editor-ui:build: ERROR: command finished with error: command (/home/.../n8n/packages/editor-ui) pnpm run build exited (1)
command (/home/.../n8n/packages/editor-ui) pnpm run build exited (1)
```

while working fine for my colleague. Tried cleaning, re-cloning etc but no luck. 

What eventually made it work was to prune/delete all Docker containers, images and volumes.

**Comment by Joffcom** [comment]
Deleting docker containers is a bit odd for a package build 🤔

Were you low on space or resources?

**Comment by martinse** [comment]
Sorry I think I might have missed the part that this was a pure `pnpm build` on the host/dev machine.  I got this exact same error when doing a custom Docker build

```
docker build  -t n8n-custom -f docker/images/n8n-custom/Dockerfile .
```

on the step `RUN pnpm build`, thus the prune solution.

In any case, my host system was not low on resources, running OSX 13.2 on 2,3 GHz  Intel i7 with 32G RAM -- with 4 cores, 6GB RAM and 256 GB HDD allocated to Docker.

**Comment by Joffcom** [comment]
It could have been that docker ran out of memory on the build, I remember on Mac having to increase the memory available.

**Comment by bracci-quest-it** [comment]
The problem was mine, probably my pc went out of memory during the build process.
I tried with a different pc and everything goes fine.
